### PR TITLE
stm32/boards: add support for MikroElektronika Clicker 2 for STM32

### PR DESCRIPTION
**Pull request by cdwilson**
- STM32F407VGT6 (1MB of Flash, 192+4 Kbytes of SRAM)
- 5V (via USB) or Li-Polymer Battery (3.7V) power input
- 2 x LEDs
- 2 x user switches
- 2 x mikroBUS sockets
- 2 x 1x26 mikromedia-compatible headers (52 pins)

https://www.mikroe.com/clicker-2-stm32f4

### CODE REVIEW

**Comment by cdwilson** [comment]
One thing I'm still a little unclear on: the documentation says that for bare metal, sys.platform should describe the hardware that micropython is running on.  In this PR, it's set to the string below, but this value is typically things like "linux", "windows", etc for python running in an OS.  Is it OK to use a descriptive string for this, or should it be something more concise (without spaces, one word, etc)?

```
#define MICROPY_PY_SYS_PLATFORM     "Mikroe Clicker 2 for STM32"
```

**Comment by dpgeorge** [comment]
Thanks for the addition, looks good.

> One thing I'm still a little unclear on: the documentation says that for bare metal, sys.platform should describe the hardware that micropython is running on.

For stm32-based boards, all of them use the value of "pyboard" for `sys.platform`, so I suggest to keep it that way for this new board as well (ie just delete the corresponding line in the board config and let the default value from `ports/stm32/mpconfigport.h` be used).

esp8266 boards use "esp8266", esp32 boards use "esp32", the zephyr port uses "zephyr".  Arguable the stm32 port should use "stm32" but it's a bit hard to change it now (would need to think how it would affect existing code that detects based on this value).

In summary: this PR would be good to merge if the platform config was just left to the (pyboard) default.

**Comment by cdwilson** [comment]
Thanks for the clarification!  Makes sense, I've removed the `#define` in the updated commit.

**Comment by dpgeorge** [comment]
Thanks for updating, merged in f9d142523c48af1cb9c83d1dfbc320cb9c2000c6 with some minor formatting changes to mpconfigboard.h.
